File: tests/phpunit/CalendarTest.php
<?php

namespace Ecma\Test\Intl;

use Ecma\Intl\Calendar;

use function json_encode;

class CalendarTest extends TestCase
{
    /**
     * @dataProvider calendarValuesProvider
     */
    public function testCalendarValues(Calendar $calendar, string $expectedValue): void
    {
        $this->assertSame($expectedValue, $calendar->value);
    }

    /**
     * @return array<array{calendar: Calendar, expectedValue: string}>
     */
    public function calendarValuesProvider(): array
    {
        return [
            [
                'calendar' => Calendar::Buddhist,
                'expectedValue' => 'buddhist',
            ],
            [
                'calendar' => Calendar::Chinese,
                'expectedValue' => 'chinese',
            ],
            [
                'calendar' => Calendar::Coptic,
                'expectedValue' => 'coptic',
            ],
            [
                'calendar' => Calendar::Dangi,
                'expectedValue' => 'dangi',
            ],
            [
                'calendar' => Calendar::Ethioaa,
                'expectedValue' => 'ethioaa',
            ],
            [
                'calendar' => Calendar::Ethiopic,
                'expectedValue' => 'ethiopic',
            ],
            [
                'calendar' => Calendar::Gregory,
                'expectedValue' => 'gregory',
            ],
            [
                'calendar' => Calendar::Hebrew,
                'expectedValue' => 'hebrew',
            ],
            [
                'calendar' => Calendar::Indian,
                'expectedValue' => 'indian',
            ],
            [
                'calendar' => Calendar::Islamic,
                'expectedValue' => 'islamic',
            ],
            [
                'calendar' => Calendar::IslamicCivil,
                'expectedValue' => 'islamic-civil',
            ],
            [
                'calendar' => Calendar::IslamicRgsa,
                'expectedValue' => 'islamic-rgsa',
            ],
            [
                'calendar' => Calendar::IslamicTbla,
                'expectedValue' => 'islamic-tbla',
            ],
            [
                'calendar' => Calendar::IslamicUmalqura,
                'expectedValue' => 'islamic-umalqura',
            ],
            [
                'calendar' => Calendar::Iso8601,
                'expectedValue' => 'iso8601',
            ],
            [
                'calendar' => Calendar::Japanese,
                'expectedValue' => 'japanese',
            ],
            [
                'calendar' => Calendar::Persian,
                'expectedValue' => 'persian',
            ],
            [
                'calendar' => Calendar::Roc,
                'expectedValue' => 'roc',
            ],
        ];
    }
}

File: tests/criterion/php/support/calendar_test.c
#include "src/php/support/calendar.h"
#include "tests/criterion/test.h"

#define TEST_SUITE phpSupportCalendar

ParameterizedTestParameters(TEST_SUITE, getsCalendarCaseNameForValue) {
  START_BASIC_TEST_PARAMS(19)

  BASIC_TEST("buddhist", "Buddhist")
  BASIC_TEST("chinese", "Chinese")
  BASIC_TEST("coptic", "Coptic")
  BASIC_TEST("dangi", "Dangi")
  BASIC_TEST("ethioaa", "Ethioaa")
  BASIC_TEST("ethiopic", "Ethiopic")
  BASIC_TEST("gregory", "Gregory")
  BASIC_TEST("hebrew", "Hebrew")
  BASIC_TEST("indian", "Indian")
  BASIC_TEST("islamic", "Islamic")
  BASIC_TEST("islamic-civil", "IslamicCivil")
  BASIC_TEST("islamic-rgsa", "IslamicRgsa")
  BASIC_TEST("islamic-tbla", "IslamicTbla")
  BASIC_TEST("islamic-umalqura", "IslamicUmalqura")
  BASIC_TEST("iso8601", "Iso8601")
  BASIC_TEST("japanese", "Japanese")
  BASIC_TEST("persian", "Persian")
  BASIC_TEST("roc", "Roc")
  BASIC_TEST("foo", NULL)
  BASIC_TEST(NULL, NULL)

  END_BASIC_TEST_PARAMS;
}

BASIC_PARAMETERIZED_TEST(getsCalendarCaseNameForValue, getCalendarCaseName)
